### ISSUE TITLE
Invert gradient op

### ISSUE THREAD

**Issue by alexatknit** [tensorflow/tensorflow#4342]
This one should be pretty simple. It would be nice to have an op that acts as the identity function in feedforward, but during backpropagation it propagates the negative of the gradients. This is useful when joining two datasets as it allows for the suppression of features that distinguish between domains. Its referred to as a gradient reversal layer in [http://arxiv.org/pdf/1505.07818v4.pdf](http://arxiv.org/pdf/1505.07818v4.pdf)

**Comment by yaroslavvb** [comment]
You can do it by using stop_gradient --
http://stackoverflow.com/questions/36456436/how-can-i-define-only-the-gradient-for-a-tensorflow-subgraph/36480182#36480182

**Comment by alexatknit** [comment]
Nice!

**Comment by andydavis1** [comment]
Thanks Yaroslav. Closing this out.
